firmware/ventilator-controller-stm32/Core/Inc/Pufferfish/Util/Vector.tpp

```diff
@@ -69,28 +69,33 @@ constexpr Element &Vector<Element, array_size>::operator[](size_t position) {
 
 template <typename Element, size_t array_size>
 template <size_t source_size>
-void Vector<Element, array_size>::copy_from(
+IndexStatus Vector<Element, array_size>::copy_from(
     const std::array<Element, source_size> &source_bytes, size_t dest_start_index) {
-  copy_from(source_bytes.data(), source_bytes.size(), dest_start_index);
+  return copy_from(source_bytes.data(), source_bytes.size(), dest_start_index);
 }
 
 template <typename Element, size_t array_size>
-void Vector<Element, array_size>::copy_from(
+IndexStatus Vector<Element, array_size>::copy_from(
     const Vector<Element, array_size> &source_bytes, size_t dest_start_index) {
-  copy_from(source_bytes.buffer(), source_bytes.size(), dest_start_index);
+  return copy_from(source_bytes.buffer(), source_bytes.size(), dest_start_index);
 }
 
 template <typename Element, size_t array_size>
-void Vector<Element, array_size>::copy_from(
+IndexStatus Vector<Element, array_size>::copy_from(
```

Comment:
Pseudocode for copy_from:
check if we have enough capacity; if not, return early and don't change any internal state: if (source_size + dest_start_index > array_size) return out_of_bounds
resize to the exact size after copy: size_ = source_size + dest_start_index
do the memcpy